gitter.py

```diff
@@ -297,9 +300,14 @@ def background():
                 else:
                     log.debug('Received keep-alive on %s', room.name)
 
-        t = threading.Thread(target=background)
-        t.daemon = True
-        t.start()
+        if room._uri not in self._joined_rooms:
+            t = threading.Thread(target=background)
+            t.daemon = True
+            t.start()
+            with self._joined_rooms_lock:
+                self._joined_rooms.append(room._uri)
```

Comment:
This is still incorrect. You're still accessing `self._joined_rooms` above, without the lock held. I'll modify it to show the fix.